daffodil-sapi/src/test/resources/test/sapi/mySchemaStringAsXml.dfdl.xsd

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<schema xmlns="http://www.w3.org/2001/XMLSchema"
+  targetNamespace="http://example.com"
+  xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/"
+  xmlns:dfdlx="http://www.ogf.org/dfdl/dfdl-1.0/extensions"
+  xmlns:xs="http://www.w3.org/2001/XMLSchema"
+  xmlns:tns="http://example.com">
+
+  <include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    
+  <annotation>
+    <appinfo source="http://www.ogf.org/dfdl/">
+      <dfdl:format ref="tns:GeneralFormat" />
+    </appinfo>
+  </annotation>
+  
+  <element name="file" type="xs:string"
```

Comment:
This test shows the feature working in a minimalist way, but it is not useful as an example of what this feature is really intended to enable. 

I can approve this PR as is, and we can make creating the more illustrative test a separate ticket, or we can just create it now. 

I would like the example to start with  src/test/resources/xmlPayloadSchemaLocation/xmlPayload.xsd, the schema for the payload XML. This schema must be able to be reused without modification. This needs to have multiple attributes declared as well as elements.

Then there is the DFDL schema binMessage.dfdl.xsd, which describes a file containing an **array** of binary data records each with four child elements: someNum, strLen, xmlStr, and anotherNum. strLen should be the length of the string. xmlStr should have an explicit simple type that extends xs:string and has a pattern facet that will fail if validation is limited (since limited validation happens during parsing, so before any of this XML trickery takes place). 

binMessageWithXML.dat should be the binary file.  The XML string inside it should contain all the examples of things that will NOT be preserved by our particular form of canonicalization, and those things that are unexpected (by some people) to be preserved. 

Things we will preserve (as I understand it)
- Character Entities
- CDATA regions
- Comments
- Processing Instructions (not just the first line, but embedded within the XML)
- DTD

Things we don't preserve (as I understand it)
- single vs. double quotes
- attribute values with leading/trailing/multiple-adjacent spaces
- maybe order of attributes (not clear if that will change or be preserved)
- CRLF
- CR (isolated)
- choice of which namespace prefix is used (Test data needs to have multiple prefix defs that are for the same namespace, so data goes in with one prefix, comes out with another.)

Default namespace definitions should be used in binMessage.dfdl.xsd, and in xmlPayload.xsd so as to show this is handled properly. Some namespace prefix definitions should also be repeated (for different namespaces) in these two schemas so as to show there is no conflict. 

This XML should contain something that violates a pattern facet in xmlPayload.xsd, like a number out of range for min/maxInclusive facets. 

Then there is binMessageWithXMLPayload.xml which is the expected combined XML output, for comparison with actual. This is created by hand to match what the XMLTextInfosetOutputter produces for the combined XML. 

Then there is binMessageWithXMLPayload.xsd which is the combined XSD that can be used to validate binMessageWithXMLPayload.xml. This schema should xs:import xmlPayload.xsd, and reference its root element as a child of the xmlStr element, which is not an xs:string, but a complex type having one child, an element ref to the root of xmlPayload.xsd. Comments should explain how xmlStr's definition has been replaced, blah blah. 

It is important to illustrate (and point out in comments) that xmlPayload.xsd is reused like this  by binMessageWithXMLPayload.xsd without change. 

The scala code should illustrate 

- the DFDL parse (with 'limited' validation, which should output that the pattern isn't matched as a validation error, as a way of showing limited validation works). 
- Then the XML result of the parse should be validated with Xerces or other XSD validation tool against the combined binMessageWithXMLPayload.xsd, which should work and report a validation error (number out of range).
- Then the unparse should create output1 but the output1 should have a canonicalized XML string, and so not matching the input exactly because the xmlStr isn't the same. Comparison should show this. 
- Then we parse this output1 data again. The infoset2 XML should again match the expected.
- Then we unparse infoset2  and the binary output2 shoud match output1 byte for byte. This illustrates that the canonicalization preserved everything that can be preserved about the xmlStr element. 

A second test should use a binary data file containing 2 records, not just one. The XML can be canonical in those, and  we don't need it to cause the validation errors. It's just showing that the inputter/outputter are managing state right, and can handle data that has multiple of these stringAsXML strings in it.

Reply:
Yeah, I'll make some tests as suggestion to make sure was test as much of this as possible.